cmd/test.go

```diff
@@ -0,0 +1,290 @@
+// Copyright © 2022-2024 Obol Labs Inc. Licensed under the terms of a Business Source License 1.1
+
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	"github.com/obolnetwork/charon/app/errors"
+)
+
+type testConfig struct {
+	OutputFormat string
+	OutputFile   string
+	Quiet        bool
+	TestCases    []string
+	Timeout      time.Duration
+}
+
+func newTestCmd(cmds ...*cobra.Command) *cobra.Command {
+	root := &cobra.Command{
+		Use:   "test",
+		Short: "Test subcommands provide test suite to evaluate current cluster setup",
+		Long:  `Test subcommands provide test suite to evaluate current cluster setup. Currently there is support for peer connection tests, beacon node and validator API.`,
+	}
+
+	root.AddCommand(cmds...)
+
+	return root
+}
+
+func bindTestFlags(cmd *cobra.Command, config *testConfig) {
+	cmd.Flags().StringVar(&config.OutputFile, "output-file", "", "File path to which output can be written.")
+	cmd.Flags().StringVar(&config.OutputFormat, "output-format", "json", "File format to which output is written. Flag --output-file is required.")
+	cmd.Flags().StringSliceVar(&config.TestCases, "test-cases", nil, "List of comma separated names of tests to be exeucted.")
+	cmd.Flags().DurationVar(&config.Timeout, "timeout", 24*time.Hour, "Execution timeout for all tests.")
+	cmd.Flags().BoolVar(&config.Quiet, "quiet", false, "Do not print test results to stdout.")
+}
+
+type Duration struct {
```

Comment:
This might be useful for other commands, I suggested to extract this into a separate file and add _test.go

Reply:
Moved it to `duration.go`. I will push tests for it alongside with the other test files